service/src/main/java/bio/terra/workspace/common/utils/GcpUtils.java

```diff
@@ -148,4 +150,22 @@ public static GoogleCredentials getGoogleCredentialsFromUserRequest(
     AccessToken accessToken = new AccessToken(userRequest.getRequiredToken(), null);
     return GoogleCredentials.create(accessToken);
   }
+
+  /**
+   * Extract the region part from the given location string. If the string is a region, return that.
+   * If the string looks like a zone, return just the region part.
+   */
+  public static String parseRegion(String location) {
+    if (location.length() <= 2) {
+      return location;
+    }
+
+    Pattern pattern = Pattern.compile("-[a-z]$", Pattern.CASE_INSENSITIVE);
```

Comment:
nit: zonePattern

i think you can merge l164-166 into one using java's replaceAll with regex? 
location.replaceAll("(?!^)-[a-z]$", "");
https://www.baeldung.com/java-remove-trailing-characters

Reply:
that looks nicer, thanks.